src/lang/eval/EvalUtil.ml

```diff
@@ -238,20 +222,15 @@ module Configuration = struct
 
   (*  Manipulations with configuartion *)
   
-  let store st k v =
-    match v with 
-    | Env.ValClosure _ | Env.ValFix _ | Env.ValTypeClosure _ ->
-        fail @@ sprintf "Cannot store a closure below into a field %s:\n%s"
-          k (Env.pp_value v)
-    | Env.ValLit l ->
-        (let s = st.fields in
-         match List.find s ~f:(fun (z, _) -> z = k) with
-         | Some (_, l') -> pure @@
-             ({st with
-              fields = (k, l) :: List.filter ~f:(fun z -> fst z <> k) s}
-              , G_Store(l', l))
-         | None -> fail @@ sprintf
-               "No field \"%s\" in fields:\n%s" k (pp_literal_map s))
+  let store st k l =
+    let s = st.fields in
+    match List.find s ~f:(fun (z, _) -> z = k) with
+    | Some (_, l') -> pure @@
```

Comment:
Should this be `Some (_, l') when is_pure_literal l' -> pure @@ ...` or do we rely on the static checker for it?